### PR TITLE
Jsdoc

### PR DESCRIPTION
**Pull request by cflynn07**
just rereviewing docker-listener before I embark on a journey to debug the container-die log event not reaching the client. Comments and file cleanup as I read each file.

Swapped lib/loadenv for @rsandor 's loadenv npm module

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
+/**
+ * create shared instance of express & set up basic status response HTTP routes
+ * @module lib/app
+ */
 'use strict';
-require('./loadenv')();
```
Comment (anandkumarpatel) [inline lib/app.js:2]: removed but not added?

File: lib/docker.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
+/**
+ * @module lib/docker
+ */
 'use strict';
-require('../lib/loadenv')();
```
Comment (anandkumarpatel) [inline lib/docker.js:2]: removed but not added?

File: lib/events.js [hunk 3]
```diff
@@ -1,19 +1,24 @@
+/**
+ * @module lib/events
+ */
 'use strict';
-require('loadenv.js')();
```
Comment (anandkumarpatel) [inline lib/events.js:2]: removed but not added?

File: lib/listener.js [hunk 4]
```diff
@@ -1,19 +1,24 @@
+/**
+ * Listen to Docker events and publish/proxy to Redis
+ * @module lib/listener
+ */
 'use strict';
-require('./loadenv')();
```
Comment (anandkumarpatel) [inline lib/listener.js:2]: removed but not added?

File: lib/publisher.js [hunk 5]
```diff
@@ -1,18 +1,27 @@
+/**
+ * @module lib/publisher
+ */
 'use strict';
-require('./loadenv')();
```
Comment (anandkumarpatel) [inline lib/publisher.js:2]: removed but not added?

File: test/publisher.js [hunk 6]
```diff
@@ -1,19 +1,23 @@
+/**
+ * @module test/publisher
+ */
 'use strict';
-var Code = require('code');
 
+require('loadenv')();
```
Comment (rsandor) [inline test/publisher.js]: No debug scope?

File: test/listener.js [hunk 7]
```diff
@@ -1,26 +1,30 @@
+/**
+ * @module test/listener
+ */
 'use strict';
-var Code = require('code');
 
+require('loadenv')();
```
Comment (rsandor) [inline test/listener.js]: Do we not need a debug scope here?

File: test/events.js [hunk 8]
```diff
@@ -1,19 +1,24 @@
+/**
+ * @module lib/events
+ */
 'use strict';
-require('../lib/loadenv')();
-var Code = require('code');
 
+require('loadenv')();
```
Comment (rsandor) [inline test/events.js]: No debug scope?

File: test/app.js [hunk 9]
```diff
@@ -1,13 +1,18 @@
+/**
+ * @module test/app
+ */
 'use strict';
-var Code = require('code');
 
+require('loadenv')();
```
Comment (rsandor) [inline test/app.js]: No debug scope?

File: index.js [hunk 10]
```diff
@@ -1,6 +1,11 @@
+/**
+ * @module index
+ */
 'use strict';
-require('./lib/loadenv')();
-var main = require('./main.js');
+
+require('loadenv')('docker-listener');
```
Comment (rsandor) [inline index.js]: Should we use `('docker-listener:env')` instead?

File: lib/publisher.js (hunk 5 above)
Comment (cflynn07) [inline lib/publisher.js:2]: yes, no need to load it from every file. Once from the index file is enough. DRY

File: lib/listener.js (hunk 4 above)
Comment (cflynn07) [inline lib/listener.js:2]: see comment below

File: lib/events.js (hunk 3 above)
Comment (cflynn07) [inline lib/events.js:2]: comment below

File: lib/docker.js (hunk 2 above)
Comment (cflynn07) [inline lib/docker.js:2]: comment below

File: lib/app.js (hunk 1 above)
Comment (cflynn07) [inline lib/app.js:2]: comment below

File: lib/publisher.js (hunk 5 above)
Comment (rsandor) [inline lib/publisher.js:2]: I was under the impression we should include it for increased modularity, is this incorrect? For instance, you could load a single module in a unit test and be safe without having to include `loadenv` in the test itself.

Comment (cflynn07) [inline lib/publisher.js:2]: We're not doing this in any other repo, and it's unnecessary. Loading once from the index file is cleaner/simpler

Comment (podviaznikov) [inline lib/publisher.js:2]: I was including `loadenv` in each file for testing. Basically in that way you can see explicitly what are requirements for the current module (some libraries and some envs). So if module needs envs it's explicit.

But I'm not against loading envs just once.

Comment (cflynn07) [inline lib/publisher.js:2]: I moved the require statements into the tests, which I think makes more sense. loading the env's from the test files makes sense for testing, no need to rerun code for each module in non-test running mode

**Comment by cflynn07** [comment]
ready to merge?

**Comment by cflynn07** [comment]
per discussion morning of April 10th (Friday) will be readding loadenv to each module

**Pull request merged**